modules/gdscript/gdscript.cpp

```diff
@@ -2141,8 +2103,27 @@ void GDScriptLanguage::remove_named_global_constant(const StringName &p_name) {
 	named_globals.erase(p_name);
 }
 
+void GDScriptLanguage::thread_enter() {
+	GDScript::func_ptrs_to_update_thread_local = memnew(GDScript::UpdatableFuncPtr);
+	GDScript::func_ptrs_to_update_thread_local->mutex.lock();
+	GDScript::func_ptrs_to_update_thread_local->mutex.unlock();
```

Comment:
Is this a leftover or this is intentional? Maybe adding a comment here explaining the point of locking and unlocking the mutex without doing anything.

Reply:
I added that deliberately to try to account for a possibility, but it doesn't make sense, so I've removed it.

I wanted to make sure that, by the unlock, the recently allocated object's bytes are effectively published to other threads that already exist and may need to lock this mutex. However, that wouldn't cover the mutex itself. Therefore, if that situation happens to be possible, we would be in a bad situation regardless any desperate attempt like mine. One situation where that may happen is the following:
- Thread 1 is created.
- Thread 2 is created.
- On thread 2, `GDScript::_add_func_ptr_to_update()`. The script takes awareness of thread 1's mutex (adds thread 1's `UpdatableFuncPtr` to its `func_ptrs_to_update`).
- On thread 1, `GDScript::clear()` is called on the same script. It will lock thread 2's mutex. Thread 1 may not have had a chance yet to see thread 2's `UpdatableFuncPtr`, which includes the mutex (so it may even cause a crash).

I'm trying to reason if that can ever happen. All in all, that's separate from this PR.